### PR TITLE
fix!: improve exports and abstract class

### PR DESCRIPTION
**Pull request by ethanve**
Mostly just to bump to v1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
   ],
   "scripts": {
     "build": "4c build src",
-    "format": "4c format src './*'",
-    "lint": "4c lint src './*'",
+    "format": "4c format src './*' --prettier-ignore .eslintignore",
+    "lint": "4c lint src './*' --prettier-ignore .eslintignore",
```
Comment (taion) [inline package.json:20]: **nitpick:** i think? no need to whitelist paths if using an ignore file

```suggestion
    "format": "4c format --prettier-ignore .eslintignore .",
    "lint": "4c lint --prettier-ignore .eslintignore .",
```

File: package.json [hunk 2]
```diff
@@ -104,4 +104,4 @@
   "readme": "ERROR: No README data found!",
   "homepage": "https://github.com/4Catalyzer/graphql-subscription-server#readme",
   "_id": "@4c/graphql-subscription-server@1.0.0"
```
Comment (taion) [inline package.json:106]:
```suggestion
```

**Comment by ethanve** [comment]
@taion any idea?
```
/home/travis/build/4Catalyzer/graphql-subscription-server/test/AsyncQueue.test.ts
  1:1  error  Definition for rule 'import/no-extraneous-dependencies' was not found  import/no-extraneous-dependencies
/home/travis/build/4Catalyzer/graphql-subscription-server/test/RedisSubscriber.test.ts
  1:1  error  Definition for rule 'import/no-extraneous-dependencies' was not found  import/no-extraneous-dependencies
/home/travis/build/4Catalyzer/graphql-subscription-server/test/__mocks__/redis.ts
  1:1  error  Definition for rule 'import/no-extraneous-dependencies' was not found  import/no-extraneous-dependencies
```

**Comment by ethanve** [comment]
the package.json does have `eslint-plugin-import` as a dep

**Comment by taion** [comment]
it's because the config is missing the `import` plugin. why? because...

`4catalyzer-typescript` works differently from `4catalyzer-flow` – it's an add-on to an existing config, like w/ https://github.com/4Catalyzer/graphql-node-resource/blob/1566ebc6cd842bc9617c2fc4b3fde2ab1ad9a294/.eslintrc#L4

so add back `eslint-config-4catalyzer`

**Pull request merged**